core/src/main/scala/org/apache/spark/internal/config/ConfigBuilder.scala

```diff
@@ -104,7 +104,7 @@ private[spark] class TypedConfigBuilder[T](
   /** Checks if the user-provided value for the config matches the validator. */
   def checkValue(validator: T => Boolean, errorMsg: String): TypedConfigBuilder[T] = {
     transform { v =>
-      if (!validator(v)) throw new IllegalArgumentException(errorMsg)
+      if (!validator(v)) throw new IllegalArgumentException(s"'$v' is invalid. $errorMsg")
```

Comment:
This looks clearer. btw, nit: how about `s"'$v' is invalid because: $errorMsg"`

Reply:
sgtm